modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderFactory.java

```diff
@@ -125,4 +127,19 @@ private void injectValueProviderFields(ValueProvider resolver) throws ValueResol
           + missingParameters, MISSING_REQUIRED_PARAMETERS);
     }
   }
+
+  private ValueProvider adaptResolver(Object resolverObject) throws ValueResolvingException {
+    if (resolverObject instanceof ValueProvider) {
+      return (ValueProvider) resolverObject;
+    } else if (resolverObject instanceof org.mule.runtime.extension.api.values.ValueProvider) {
```

Comment:
wrong adapter direction

Reply:
This is the opposite of what I did in the previous commits. where I was adapting the other way and we agree that it was wrong.

Reply:
yes, sorry, got confused by the classnames. `SdkValueProviderAdapter` should be called `LegacyValueProviderAdapter` ?